File: week-2/counting.py
def countingSort(lst):
    max = 0
    sortedArray = []
    for i in range(len(lst)):
        if(lst[i] > max):
            max = lst[i]
    countArray = [0] * (max + 1)
    for i in range(len(lst)):
        countArray[lst[i]] += 1
    for i in range(len(countArray)):
        for j in range (countArray[i]):
            sortedArray.append(i)
    print(sortedArray)
    return sortedArray

countingSort([1, 2, 3, 1, 3])

File: week-2/counting.js
function countingSort(ar) {
    let max = 0;
    let countArray = [];
    let sortedArray = [];
    for(let i = 0; i <= ar.length - 1; i++) {
        if(ar[i] > max) {
            max = ar[i];
        }
    }
    for(let i = 0; i <= max; i++) {
        countArray.push(0);
    }
    for(let i = 0; i <= ar.length - 1; i++) {
        countArray[ar[i]] += 1;
    }
    for(let i = 0; i <= countArray.length - 1; i++) {
        for(let j = 0; j < countArray[i]; j++) {
            sortedArray.push(i);
        }
    }
    console.log(sortedArray);

}

countingSort([4, 3, 2, 1, 1]);
